sound/soc/sof/sof-client-probes-ipc4.c

```diff
@@ -170,16 +175,55 @@ static int ipc4_probes_deinit(struct sof_client_dev *cdev)
  * @desc:	Returned list of active probes
  * @num_desc:	Returned count of active probes
  * @return:	0 on success, negative error code on error
- *
- * Dummy implementation returning empty list of probes.
  */
 static int ipc4_probes_points_info(struct sof_client_dev *cdev,
 				   struct sof_probe_point_desc **desc,
 				   size_t *num_desc)
 {
-	/* TODO: Firmware side implementation needed first */
-	*desc = NULL;
-	*num_desc = 0;
+	struct sof_man4_module *mentry = sof_ipc4_probe_get_module_info(cdev);
+	struct device *dev = &cdev->auxdev.dev;
+	struct sof_ipc4_probe_info *info;
+	struct sof_ipc4_msg msg;
+	int i, ret;
+
+	if (!mentry)
+		return -ENODEV;
+
+	msg.primary = mentry->id;
+	msg.primary |= SOF_IPC4_MSG_DIR(SOF_IPC4_MSG_REQUEST);
+	msg.primary |= SOF_IPC4_MSG_TARGET(SOF_IPC4_MODULE_MSG);
+
+	msg.extension = SOF_IPC4_MOD_EXT_MSG_PARAM_ID(SOF_IPC4_PROBE_POINTS);
+
+	msg.data_size = cdev->sdev->ipc->max_payload_size;
+	msg.data_ptr = kzalloc(msg.data_size, GFP_KERNEL);
+	if (!msg.data_ptr)
+		return -ENOMEM;
+
+	ret = sof_client_ipc_set_get_data(cdev, &msg, false);
+	if (ret) {
+		dev_err(dev, "%s: ipc failed %u", __func__, ret);
```

Comment:
Ok, I'll make this a debug.

Reply:
It might be OK to drop the print altogether? The core will print error anyways.

Reply:
But, but. If this fails once, the chances to fail next time is pretty high, right? Should we disable the probes or fallback to what we had before?

Reply:
There was nothing before. The callback just returned an empty array of probe points, so the only difference to the earlier behavior is the error message.

Reply:
@jsarha, the difference is that this will introduce regression if you update the kernel w/o touching the probes, what worked will no longer works.
Albeit, it is a low impact regression, the probes are not used too frequently.

Reply:
@ujfalusi there is some new error messages in the logs when reading "probe_points" debugfs or when sof_probes_compr_shutdown() is called, but there is no other difference in behavior. I'd say very low impact regression, as the only annoyance is couple of new error messages.